design/one-pager-release-process.md

```diff
@@ -0,0 +1,321 @@
+the [semver spec](https://semver.org/#spec-item-4), described as such:
+
+> Major version zero (0.y.z) is for initial development. Anything MAY change at any time. The public
+> API SHOULD NOT be considered stable.
+
+In the meantime, every monthly release will be a **minor** release, meaning we will be incrementing
+the minor version number. For example, the next 3 monthly releases of Crossplane will be `v0.4.0`,
+`v0.5.0`, and `v0.6.0`.
+
+### Patch Releases
+
+In addition to monthly minor releases, we will also be able to perform patch releases in the event
+that a severe issue is identified in the last release. Patch releases will increment the `PATCH`
+version field, for example a patch for the `v0.4.0` release will have a version number of `v0.4.1`.
+
+The exact criteria for what justifies performing a patch release is difficult to identify since
+every issue is unique. However, the team will perform triage on any issues reported in the latest
+release. In general, a patch release will be justified if an issue greatly affects reliability,
+stability, or usability, measured both by severity as well as frequency.
+
+The hope is that our release automation will be simple and streamlined enough that the cost of a
+patch release is fairly minimal.
+
+#### Support of Previous Versions
+
+In terms of the scope of previous releases that we will continue to support, we propose to only
+support the last/latest release. For example, when `v0.5.0` is released, we will cease to continue
+providing patch releases for `v0.4.0`. Only the last release will receive patch updates.
+
+This policy is open to change in the future as we gain more adoption and the community chooses to
+stay on older releases for longer. Any decision to change this policy will take into account user
+and community feedback.
+
+#### Patch Release Process
+
+Patch releases are always performed from the release branch of the minor release the patch is
+applicable to. For example, a patch release of `v0.4.1` would be committed to and built from the
+`release-0.4` branch.
+
+1. Issue is first fixed in `master` with the normal PR workflow
+    1. If the issue doesn't exist in `master`, perhaps because the offending code has been removed
+       since the last release, then the fix PR should be to the release branch
+1. A PR is opened in the release branch that contains a cherry-pick of the **fix commit** (not the
+   merge commit)
+1. The cherry-pick PR is merged into the release branch
+1. The patch release pipeline automation is run with an incremented `PATCH` field from the last
+   release, e.g. `v0.4.1`.
+
+## Golden Master
+
+To facilitate the quality of releases and the ease of performing a release, we will be adopting a
+"golden master" philosophy for the `master` branch. Essentially, all changes that are accepted and
+merged into `master` should be **complete** and of high quality. In addition to the functional code
```

Comment:
To be able to move forward fast, it may be helpful to permit merges for partial features which have no breaking changes, no side-effects, and can only be accessed deliberately.  This could leave the new feature without documentation unless the onus is passed to the PR that exposes the feature.

For example, @suskin @turkenh and I were considering introducing the Template Stacks harnessing in piece-meal PRs rather than one larger PR later.

How would you characterize that endeavour as it relates to the golden master philosophy?

Reply:
Yes that's a good point @displague to be able to include features that must be deliberately accessed.  We don't have a feature gate functionality right now, but you could consider behavior requiring deliberate commands from the user (as opposed to behavior on mainline paths) to be similar to a feature gate.  I'll include a note on this.